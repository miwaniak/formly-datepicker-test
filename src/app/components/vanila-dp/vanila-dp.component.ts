import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-vanila-dp',
  templateUrl: './vanila-dp.component.html',
  styleUrls: ['./vanila-dp.component.scss']
})
export class VanilaDpComponent implements OnInit {

  constructor() { }

  ngOnInit() {
  }

}
