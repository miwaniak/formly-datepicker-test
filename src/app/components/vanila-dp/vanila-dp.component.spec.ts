import { async, ComponentFixture, TestBed } from "@angular/core/testing";

import { VanilaDpComponent } from "./vanila-dp.component";
import { CommonModule } from "@angular/common";
import { ReactiveFormsModule } from "@angular/forms";
import {
  MatInputModule,
  MatDatepickerModule,
  MatNativeDateModule
} from "@angular/material";
import { NoopAnimationsModule } from "@angular/platform-browser/animations";

describe("VanilaDpComponent", () => {
  let component: VanilaDpComponent;
  let fixture: ComponentFixture<VanilaDpComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        CommonModule,
        ReactiveFormsModule,
        MatInputModule,
        MatDatepickerModule,
        MatNativeDateModule,
        NoopAnimationsModule
      ],
      declarations: [VanilaDpComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(VanilaDpComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it("should create", () => {
    expect(component).toBeTruthy();
  });

  it("should render date picker", () => {
    expect(fixture).toMatchSnapshot();
  });
});
