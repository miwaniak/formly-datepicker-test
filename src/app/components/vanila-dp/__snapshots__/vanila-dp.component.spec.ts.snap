// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VanilaDpComponent should render date picker 1`] = `
<app-vanila-dp>
  <mat-form-field
    class="mat-form-field ng-tns-c3-0 mat-primary mat-form-field-type-mat-input mat-form-field-appearance-legacy mat-form-field-can-float mat-form-field-has-label mat-form-field-hide-placeholder _mat-animation-noopable"
  >
    <div
      class="mat-form-field-wrapper"
    >
      <div
        class="mat-form-field-flex"
      >
        
        
        <div
          class="mat-form-field-infix"
        >
          <input
            aria-haspopup="true"
            aria-invalid="false"
            aria-required="false"
            class="mat-input-element mat-form-field-autofill-control cdk-text-field-autofill-monitored"
            id="mat-input-1"
            matinput=""
            ng-reflect-mat-datepicker="[object Object]"
            ng-reflect-placeholder="Choose a date"
            placeholder="Choose a date"
          />
          <mat-datepicker />
          <span
            class="mat-form-field-label-wrapper"
          >
            
            <label
              aria-owns="mat-input-1"
              class="mat-form-field-label ng-tns-c3-0 mat-empty mat-form-field-empty ng-star-inserted"
              for="mat-input-1"
              id="mat-form-field-label-3"
              ng-reflect-disabled="true"
              ng-reflect-ng-switch="false"
            >
              
              
              Choose a date
              
              
            </label>
          </span>
        </div>
        
        <div
          class="mat-form-field-suffix ng-tns-c3-0 ng-star-inserted"
        >
          <mat-datepicker-toggle
            class="mat-datepicker-toggle"
            matsuffix=""
            ng-reflect-datepicker="[object Object]"
            tabindex="-1"
          >
            <button
              aria-haspopup="true"
              aria-label="Open calendar"
              class="mat-icon-button _mat-animation-noopable"
              mat-icon-button=""
              ng-reflect-disabled="false"
              tabindex="0"
              type="button"
            >
              <span
                class="mat-button-wrapper"
              >
                
                <svg
                  class="mat-datepicker-toggle-default-icon ng-star-inserted"
                  fill="currentColor"
                  focusable="false"
                  height="24px"
                  viewBox="0 0 24 24"
                  width="24px"
                >
                  <path
                    d="M19 3h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zM7 10h5v5H7z"
                  />
                </svg>
              </span>
              <div
                class="mat-button-ripple mat-ripple mat-button-ripple-round"
                matripple=""
                ng-reflect-centered="true"
                ng-reflect-disabled="false"
                ng-reflect-trigger="[object HTMLButtonElement]"
              />
              <div
                class="mat-button-focus-overlay"
              />
            </button>
          </mat-datepicker-toggle>
        </div>
      </div>
      
      <div
        class="mat-form-field-underline ng-tns-c3-0 ng-star-inserted"
      >
        <span
          class="mat-form-field-ripple"
        />
      </div>
      <div
        class="mat-form-field-subscript-wrapper"
        ng-reflect-ng-switch="hint"
      >
        
        
        <div
          class="mat-form-field-hint-wrapper ng-tns-c3-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
          style="opacity:1;transform:translateY(0%);0:opacity;1:transform;"
        >
          
          <div
            class="mat-form-field-hint-spacer"
          />
        </div>
      </div>
    </div>
  </mat-form-field>
</app-vanila-dp>
`;
